gcc/rust/resolve/rust-late-name-resolver-2.0.cc

```diff
@@ -287,6 +287,16 @@ Late::visit (AST::TypePath &type)
   DefaultResolver::visit (type);
 }
 
+void
+Late::visit (AST::Trait &trait)
+{
+  // kind of weird how this is done
```

Comment:
very nitpicky but could you add a comment here explaining the weirdness? will be helpful for future contributors